inausoft.netCLI.all/inausoft.netCLI/MateuszCommands/MateuszCommandHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+
+namespace inausoft.netCLI.MateuszCommands
+{
+    public class MateuszCommandHandler : IMateuszCommandHandler<MateuszCommandHandler.MateuszCommand>
+    {
+
+        private MateuszCommand command;
+
+        public MateuszCommandHandler()
+        {
+            command = new MateuszCommand();
+        }
+
+        public MateuszCommand GetCommand()
+        {
+            return command;
+        }
+
+        public int Run()
```

Comment:
Take a look at this method. It is not atomic. How would you test it - without your `RootCommandHandler`?
```
var command = handler.GetCommand();
command.SampleProperty = True;

var command = new SampleCommand()
{
    SampleProperty = True
}
```

First example already introduces uncertainity about how was command created. Was the handler implemented to always created new instance or as in example was it it's own member??

Reply:
I don't think that `Command` should be instantiated outside that Handler class, therefore it is nested class. I probably should introduce Singleton patter inside Command.

I understand your point - the command properties are set outside in the `RootCommandHandler`. But should any handler be created outside that `RootCommandHandler`? I think the answer is no, therefore testing it via rootCommandHandler it is not a bad idea.

Reply:
I think you missed the point. Following line is missleading for the user.
```
var command = handler.GetCommand();
```
User don't know if the command was created at the time of invoking the method or was it initialized before. Will you get reference to the same command calling this method again or not? You can never be sure - and worst you can implement it differently each time.
On the flip side if you would initialize command outside you hvae better control - especially in your unit tests. When you don't care about the framework at all.

Reply:
I think that i got your point. Initially, my point was to get rid of `Command` object from `Run` method, but after consideration your agrument, the command needs to be passed there.